### PR TITLE
Masked phone numbers

### PR DESCRIPTION
**Pull request by theido**
https://app.asana.com/0/242012221312579/280826432931829

### CODE REVIEW

File: sources/model/managars/GGHTTPClientManager.h [hunk 1]
```diff
@@ -115,11 +115,13 @@ withCompletionHandler:(nullable GGOrderResponseHandler)completionHandler;
  *  @param findmeConfig        findme configuration object holding token and url
  *  @param lat               latitude location
  *  @param lng               longitude location
- *  @param completionHandler remote handler
+ *  @param completionHandler responce handler
```
Comment (orxelm) [inline sources/model/managars/GGHTTPClientManager.h]: typo: response

File: sources/model/managars/GGHTTPClientManager.m [hunk 2]
```diff
@@ -568,7 +569,25 @@ - (void)sendFindMeRequestWithFindMeConfiguration:(nonnull GGFindMe *)findmeConfi
...
+    NSMutableDictionary *params = [NSMutableDictionary dictionary];
+    [self addAuthinticationToParams:&params];
+    [self httpRequestWithMethod:BCRESTMethodGet
+                           path:[NSString stringWithFormat:API_PATH_SHARED_MASK_PHONE, uuid]
+                         params:params
+              completionHandler:^(BOOL success, id JSON, NSError *error) {
+                  
+                  // update last date
+                  self.lastEventDate = [NSDate date];
+                  NSString* phoneNumber = [JSON objectForKey:@"phone_number"];
```
Comment (orxelm) [inline sources/model/managars/GGHTTPClientManager.m]: i would use the JSON util here to save crashes when that value will not be a String

File: sources/model/managars/GGTrackerManager.m [hunk 3]
```diff
@@ -747,6 +747,45 @@ - (void)sendFindMeRequestForOrder:(nonnull GGOrder *)order latitude:(double)lat
...
+        if (completionHandler) {
+            
+            completionHandler(NO,nil, [NSError errorWithDomain:kSDKDomainSetup code:GGErrorTypeHTTPManagerNotSet userInfo:@{NSLocalizedDescriptionKey:@"http manager is not set"}]);
+        }
+        
+        return;
+    }
+    
+    
+    if (!uuid) {
```
Comment (orxelm) [inline sources/model/managars/GGTrackerManager.m]: missing test for this case

File: sources/model/managars/GGHTTPClientManager.h [hunk 4]
```diff
@@ -115,11 +115,13 @@ withCompletionHandler:(nullable GGOrderResponseHandler)completionHandler;
...
  */
 - (void)sendFindMeRequestWithFindMeConfiguration:(nonnull GGFindMe *)findmeConfig latitude:(double)lat longitude:(double)lng  withCompletionHandler:(nullable GGActionResponseHandler)completionHandler;
 
-
+-(void)sendMaskedNumberRequestForOrderWithUUID:(NSString *_Nonnull)uuid
```
Comment (orxelm) [inline sources/model/managars/GGHTTPClientManager.h]: missing test

Comment (theido) [inline sources/model/managars/GGHTTPClientManager.h]: please see testGetMaskedPhoneNumberWithUDID in BringgTrackingClientTests

File: sources/model/managars/GGHTTPClientManager.m (hunk 2 above)
Comment (theido) [inline sources/model/managars/GGHTTPClientManager.m]: 👍

File: sources/model/managars/GGTrackerManager.m (hunk 3 above)
Comment (theido) [inline sources/model/managars/GGTrackerManager.m]: please see testGetMaskedPhoneNumberWithNoUDID

Comment (orxelm) [inline sources/model/managars/GGTrackerManager.m]: sorry my bad, i was referring to different case, see previous and next comments

File: sources/model/managars/GGTrackerManager.m [hunk 5]
```diff
@@ -747,6 +747,45 @@ - (void)sendFindMeRequestForOrder:(nonnull GGOrder *)order latitude:(double)lat
     [self.httpManager sendFindMeRequestWithFindMeConfiguration:order.sharedLocation.findMe latitude:lat longitude:lng withCompletionHandler:completionHandler];
 }
 
+-(void)sendMaskedNumberRequestForOrderWithUUID:(NSString *_Nonnull)uuid
+                                forPhoneNumber:(NSString*_Nonnull)originalPhoneNumber
+                         withCompletionHandler:(nullable GGMaskedPhoneNumberResponseHandler)completionHandler{
+    
+    if (!self.httpManager) {
```
Comment (orxelm) [inline sources/model/managars/GGTrackerManager.m]: missing test for this case

File: sources/model/managars/GGTrackerManager.m [hunk 6]
```diff
@@ -747,6 +747,45 @@ - (void)sendFindMeRequestForOrder:(nonnull GGOrder *)order latitude:(double)lat
...
+            completionHandler(NO,nil, [NSError errorWithDomain:kSDKDomainData code:GGErrorTypeInvalidUUID userInfo:@{NSLocalizedDescriptionKey:@"supplied order uuid is invalid"}]);
+        }
+        
+        return;
+    }
+
+    GGOrder *order = [self orderWithUUID:uuid];
+    
+    if (!order) {
+        if (completionHandler) {
```
Comment (orxelm) [inline sources/model/managars/GGTrackerManager.m]: missing test for this case

**Pull request merged**